vendor/github.com/prometheus/prometheus/model/labels/labels_string.go

```diff
@@ -599,9 +599,12 @@ func (b *Builder) Get(n string) string {
 }
 
 // Range calls f on each label in the Builder.
-// If f calls Set or Del on b then this may affect what callbacks subsequently happen.
 func (b *Builder) Range(f func(l Label)) {
-	origAdd, origDel := b.add, b.del
+	// Stack-based arrays to avoid heap allocation in most cases.
+	var addStack [1024]Label
```

Comment:
I see that @bboreham has [reduced this to 128](https://github.com/prometheus/prometheus/pull/12174/files), I wonder if it's a big cost here, or we can go with 1024 and then bring the next commit?